governance/xc_admin/packages/xc_admin_frontend/components/tabs/General.tsx

```diff
@@ -0,0 +1,730 @@
+          const fileData = e.target.result
+          if (!isValidJson(fileData as string)) return
+          const fileDataParsed = sortData(JSON.parse(fileData as string))
+          const changes: Record<string, any> = {}
+          Object.keys(fileDataParsed).forEach((symbol) => {
+            if (!existingSymbols.has(symbol)) {
+              // if symbol is not in existing symbols, create new entry
+              changes[symbol] = { new: {} }
+              changes[symbol].new = { ...fileDataParsed[symbol] }
+              // these fields are generated deterministically and should not be updated
+              delete changes[symbol].new.address
+              delete changes[symbol].new.priceAccounts[0].address
+            } else if (
+              // if symbol is in existing symbols, check if data is different
+              JSON.stringify(data[symbol]) !==
+              JSON.stringify(fileDataParsed[symbol])
+            ) {
+              changes[symbol] = { prev: {}, new: {} }
+              changes[symbol].prev = { ...data[symbol] }
+              changes[symbol].new = { ...fileDataParsed[symbol] }
+            }
+          })
+          setDataChanges(changes)
+          openModal()
+        }
+      }
+      reader.readAsText(file)
+    })
+    document.body.appendChild(uploadAnchor) // required for firefox
+    uploadAnchor.click()
+    uploadAnchor.remove()
+  }
+
+  // check if uploaded json is valid json
+  const isValidJson = (json: string) => {
+    try {
+      JSON.parse(json)
+    } catch (e: any) {
+      toast.error(capitalizeFirstLetter(e.message))
+      return false
+    }
+    return true
+  }
+
+  const handleSendProposalButtonClick = async () => {
+    if (pythProgramClient && dataChanges) {
+      const instructions: TransactionInstruction[] = []
+      Object.keys(dataChanges).forEach(async (symbol) => {
+        const { prev, new: newChanges } = dataChanges[symbol]
+        // if prev is undefined, it means that the symbol is new
+        if (!prev) {
+          // deterministically generate product account key
+          const productAccountKey = await PublicKey.createWithSeed(
+            OPS_KEY,
+            'product:' + symbol,
+            pythProgramClient.programId
+          )
+          // create add product account instruction
+          instructions.push(
+            await pythProgramClient.methods
+              .addProduct()
+              .accounts({
+                fundingAccount: squads?.getAuthorityPDA(
+                  SECURITY_MULTISIG[getMultisigCluster(cluster)],
+                  1
+                ),
+                tailMappingAccount: rawConfig.mappingAccounts[0].address,
+                productAccount: productAccountKey,
+              })
+              .instruction()
+          )
+          // create update product account instruction
+          instructions.push(
+            await pythProgramClient.methods
+              .updProduct(newChanges.metadata)
```

Comment:
I found this the hard way. Don't forget that symbol lives here as well so you need to do something like:
`{...newChanges.metadata, symbol: symbol}`